### PR TITLE
Cleanup code for 0.5

### PR DESCRIPTION
**Pull request by kshyatt**
- switch to using testsets fulltime
- fix import of cudaStream_t from CUDArt.jl
- fix depwarns

@maleadt @SimonDanisch

### CODE REVIEW

File: src/libcusparse_types.jl [hunk 1]
```diff
@@ -1,10 +1,14 @@
 #enum cusparseStatus_t
 #error messages from CUSPARSE
 
+<<<<<<< HEAD
```
Comment (SimonDanisch) [inline src/libcusparse_types.jl]: merge conflict?

**Comment by kshyatt** [comment]
OK!!! This is now working on 0.5!

**Comment by SimonDanisch** [comment]
lovely! :)

**Comment by SimonDanisch** [comment]
Now we just need to figure out what to do with CUDArt -.-

**Comment by maleadt** [comment]
@SimonDanisch: I found out the bug preventing us from merging is also present in the current version, so I'll go ahead and tag (somewhere tomorrow).

**Review by SimonDanisch — APPROVED** [review]

**Pull request merged**